urllib3/util/url.py

```diff
@@ -192,7 +192,10 @@ def parse_url(url):
             # If given, ports must be integers.
             if not port.isdigit():
                 raise LocationParseError(url)
-            port = int(port)
+            try:
```

Comment:
If `port` is not an int, it will fail in the try `port = int(port)`, and raise the same error in the except block. Isn't that the same functionality?